src/python/zquantum/vqe/singlet_uccsd.py

```diff
@@ -0,0 +1,135 @@
+        Ansatz class representing Singlet UCCSD Ansatz.
+
+        Args:
+            number_of_layers: number of layers of the ansatz.
+            number_of_spatial_orbitals: number of spatial orbitals.
+            number_of_alpha_electrons: number of alpha electrons.
+            transformation: transformation used for translation between fermions and qubits.
+
+        Attributes:
+            number_of_beta_electrons: number of beta electrons (equal to number_of_alpha_electrons).
+            number_of_electrons: total number of electrons (number_of_alpha_electrons + number_of_beta_electrons).
+            number_of_qubits: number of qubits required for the ansatz circuit.
+            number_of_params: number of the parameters that need to be set for the ansatz circuit.
+        """
+        super().__init__(number_of_layers)
+        self._number_of_spatial_orbitals = number_of_spatial_orbitals
+        self._number_of_alpha_electrons = number_of_alpha_electrons
+        self._transformation = transformation
+        self._assert_number_of_spatial_orbitals()
+
+    @property
+    def number_of_spatial_orbitals(self):
+        return self._number_of_spatial_orbitals
+
+    @invalidates_parametrized_circuit
+    @number_of_spatial_orbitals.setter
+    def number_of_spatial_orbitals(self, new_number_of_spatial_orbitals):
+        self._number_of_spatial_orbitals = new_number_of_spatial_orbitals
+        self._assert_number_of_spatial_orbitals()
+
+    @property
+    def number_of_qubits(self):
+        return self._number_of_spatial_orbitals * 2
+
+    @property
+    def number_of_alpha_electrons(self):
+        return self._number_of_alpha_electrons
+
+    @invalidates_parametrized_circuit
+    @number_of_alpha_electrons.setter
+    def number_of_alpha_electrons(self, new_number_of_alpha_electrons):
+        self._number_of_alpha_electrons = new_number_of_alpha_electrons
+        self._assert_number_of_spatial_orbitals()
+
+    @property
+    def _number_of_beta_electrons(self):
+        return self._number_of_alpha_electrons
+
+    @property
+    def number_of_electrons(self):
+        return self._number_of_alpha_electrons + self._number_of_beta_electrons
+
+    @property
+    def number_of_params(self) -> int:
+        """
+        Returns number of parameters in the ansatz.
+        """
+        return uccsd_singlet_paramsize(
+            n_qubits=self.number_of_qubits, n_electrons=self.number_of_electrons,
+        )
+
+    @overrides
+    def _generate_circuit(self, params: Optional[np.ndarray] = None) -> Circuit:
+        """
+        Returns a parametrizable circuit represention of the ansatz.
+        Args:
+            params: parameters of the circuit. 
+        """
+        circuit = Circuit()
+        circuit.gates = []
+
+        # Prepare initial state
+        for i in range(self.number_of_qubits):
+            qubit_index = self.number_of_qubits - i - 1
+            circuit.gates.append(Gate("X", Qubit(qubit_index)))
```

Comment:
It seems one of the commits changed `n_electrons` back to `number_of_qubits`, thus bringing back my former comment :D This should really only apply `X` gates to the occupied qubits, which are the first `n_electrons` qubits.

Reply:
That was by accident :D 
Fixed in [f37285](https://github.com/zapatacomputing/z-quantum-vqe/pull/5/commits/f37285af8a5e296202518ac6ac42df3cb0d4bf43).